### DISCUSSION TITLE
Can I use leader key without a modifer?

### DISCUSSION THREAD

**Discussion by azzamsa** [wezterm/wezterm#1181] [Q&A] [answered]
I use one key for leader key in tmux:

```conf
set-option -g prefix `
bind-key ` send-prefix
```

So, to to split window, I can press `` ` ``  then `-`.
Is it possible in wezterm? 

Something like:

```lua
local wezterm = require 'wezterm';

return {
  -- timeout_milliseconds defaults to 1000 and can be omitted
  leader = { key="`", mods="NONE", timeout_milliseconds=1000 },
  keys = {
    {key="-", mods="LEADER", action=wezterm.action{SplitVertical={domain="CurrentPaneDomain"}}},
  }
}

```
Thanks a lot for wezterm!

**Comment by wez** [comment, marked as answer]
Yes, exactly that works!

**Reply by azzamsa** [reply]
Wow, Yes it is. I use `mods=""` multiple times and it doesn't work, the above code is just the example that pops up in my mind when I open the discussion. Turns out it is the working code.

**Comment by eugenesvk** [comment]
Is it possible to achieve the opposite — without any key (tried `key=""` and `key="NONE"`, but it didn't work). Let's say I'd like to try <kbd>Ctrl</kbd><kbd>Alt</kbd> as my leader key, how would I do that?

**Reply by wez** [reply]
Depending on the OS it *might* be possible to use `key="Ctrl"` OR `key="Alt"` to trigger when the individual key is pressed, but because those are special modifier keys they may not portably be reported as individual key presses.

It may not be possible to robustly trigger a leader key press with a chord formed from modifier keys only because there has to be a triggering key press event we can bind to, and as mentioned above, there may not be a discrete event provided by the OS, but even if there was, humans aren't super consistent with the order in which they press modifier keys and we'd need to be able to match both `key="Alt", mods="Ctrl"` and `key="Ctrl", mods="Alt"` to make things work consistently in either order.

**Reply by wez** [reply]
I'd suggest playing with https://wezfurlong.org/wezterm/config/lua/config/debug_key_events.html to see if that gives inspiration on what gets reported on your system and then see if you can apply that to your configuration

**Reply by eugenesvk** [reply]
> we'd need to be able to match both `key="Alt", mods="Ctrl"` and `key="Ctrl", mods="Alt"` to make things work consistently in either order.

Thanks, this is indeed an issue since the `Leader` key can only be assigned to one key combination,  correct? So I can't correct for this by assigning `Leader` to the same combo with different key order?

> suggest playing with https://wezfurlong.org/wezterm/config/lua/config/debug_key_events.html to see if that gives inspiration on what gets reported on your system and then see if you can apply that to your configuration

Otherwise your suggestion works beatifully, I get the events like `key: Control, modifiers: ALT | CTRL` or `key: Alt, modifiers: ALT | CTRL` and just paste them to my configs!

Ultimately, though, I thought of trying to make my <kbd>Capsclock</kbd> "Hyper" key (which is mapped to press all 4 modifiers) as a `Leader` in WezTerm, and even though I get a bunch of events with different modifier orders like so: 
```
  ... { key: LeftWindows	, modifiers:                      SUPER	, raw_key: None, raw_modifiers:                      SUPER	, raw_code: Some(91), repeat_count: 1, key_is_down: true }
  ... { key: Alt        	, modifiers:         ALT        | SUPER	, raw_key: None, raw_modifiers:         ALT        | SUPER	, raw_code: Some(18), repeat_count: 1, key_is_down: true }
  ... { key: Control    	, modifiers:         ALT | CTRL | SUPER	, raw_key: None, raw_modifiers:         ALT | CTRL | SUPER	, raw_code: Some(17), repeat_count: 1, key_is_down: true }
  ... { key: Shift      	, modifiers: SHIFT | ALT | CTRL | SUPER	, raw_key: None, raw_modifiers: SHIFT | ALT | CTRL | SUPER	, raw_code: Some(16), repeat_count: 1, key_is_down: true }
```
The last one seems to be consistent (likely due to the fact that the keys are pressed programatically rather then via an unreliable human :) ), so I might try that and see how it plays out

**Reply by wez** [reply]
> since the `Leader` key can only be assigned to one key combination, correct? So I can't correct for this by assigning `Leader` to the same combo with different key order?

Correct: wezterm only knows to look for a single leader today, so you can only specify a single leader configuration.
